### PR TITLE
docs: further improve k8s coscheduling docs

### PR DESCRIPTION
**Pull request by eecsliu**
## Description
improve the documentation around the k8s coscheduler and bump the helm chart version.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/how-to/installation/kubernetes.txt [hunk 1]
```diff
@@ -270,6 +270,22 @@ perform TLS termination in the load-balancer:
                serviceName: determined-master-service-<name for your deployment>
                servicePort: masterPort configured in values.yaml
 
+Default Scheduler (Optional)
+============================
+
+Determined includes support for the `lightweight coscheduling plugin
+<https://github.com/kubernetes-sigs/scheduler-plugins/tree/release-1.18/pkg/coscheduling>`__,
+which extends the default Kubernetes scheduler to provide gang
+scheduling. This feature is currently in beta and is not enabled by
```
Comment (justin-determined-ai) [inline docs/how-to/installation/kubernetes.txt:279]: question: What do we mean by "beta" here? Are we saying it's untested/unverified, or that we may drop support for it in the future, or something else?

File: docs/reference/helm-config.txt [hunk 2]
```diff
@@ -237,3 +237,10 @@ Helm Chart </helm/determined-helm-chart.tgz>`.
       than a path as it does in the cluster configuration. This can be
       conveniently set at the command line using ``helm install
       --set-file logging.security.tls.certificate=<path>``.
+
+-  ``defaultScheduler``: Configures the default scheduler that
+   Determined will use. Currently only supports the "coscheduler"
```
Comment (justin-determined-ai) [inline docs/reference/helm-config.txt:242]: blocking: I think we use `` for literals, not ".

File: docs/reference/helm-config.txt [hunk 3]
```diff
@@ -237,3 +237,10 @@ Helm Chart </helm/determined-helm-chart.tgz>`.
...
+   option, which enables the `lightweight coscheduling plugin
+   <https://github.com/kubernetes-sigs/scheduler-plugins/tree/release-1.18/pkg/coscheduling>`__.
+   Unless specified as "coscheduler", Determined will use the default
```
Comment (justin-determined-ai) [inline docs/reference/helm-config.txt:245]: blocking: Ditto above.

File: docs/topic-guides/system-concepts/scheduling.txt [hunk 4]
```diff
@@ -176,5 +178,10 @@ contain the following:
...
          metadata:
             labels:
-               pod-group.scheduling.sigs.k8s.io/name: determined
-               pod-group.scheduling.sigs.k8s.io/min-available: "2"
+               pod-group.scheduling.sigs.k8s.io/name: <unique task name>
+               pod-group.scheduling.sigs.k8s.io/min-available: <# of GPUs required>
+         spec:
+            schedulerName: coscheduler
+
+You can also use ``schedulerName: default-scheduler`` to use the default
```
Comment (justin-determined-ai) [inline docs/topic-guides/system-concepts/scheduling.txt:186]: question: I think I may be misunderstanding, but this is a Kubernetes value and not a master config value right?

File: helm/charts/determined/Chart.yaml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: determined
 description: A Helm chart for Determined
-version: "0.4.0"
+version: "0.5.0"
```
Comment (justin-determined-ai) [inline helm/charts/determined/Chart.yaml:4]: blocking: I think this is going to conflict with #2101. @shiyuann can you advise on what to do here?

**Review by justin-determined-ai — COMMENTED** [review]
Looks good! I think this may cause issues with one of @shiyuann's commits. Can the two of you confirm it won't or fix?

File: helm/charts/determined/Chart.yaml (hunk 5 above)
Comment (shiyuann) [reply helm/charts/determined/Chart.yaml:4]: @eecsliu yeah. you need to search for `/helm/determined-0.4.0.tgz` and bump that version in the doc.

Comment (eecsliu) [reply helm/charts/determined/Chart.yaml:4]: will do, thanks!

File: docs/how-to/installation/kubernetes.txt (hunk 1 above)
Comment (eecsliu) [reply docs/how-to/installation/kubernetes.txt:279]: I guess the closest definition would be "untested". While we have tested it a lot, I wouldn't be surprised if users found more undesirable behavior.

Comment (justin-determined-ai) [reply docs/how-to/installation/kubernetes.txt:279]: Okay, I think this is probably fine then.

File: docs/topic-guides/system-concepts/scheduling.txt (hunk 4 above)
Comment (eecsliu) [reply docs/topic-guides/system-concepts/scheduling.txt:186]: yeah you got it - the `values.yaml` is from the helm chart.

**Review by justin-determined-ai — APPROVED** [review]
Okay, I think this looks good! Confirmed same number of 0.5.0 as there were 0.4.0 in @shiyuann's PR :)

**Pull request merged**